apps/betterangels-backend/clients/tests/test_permissions.py

```diff
@@ -48,6 +61,38 @@ def test_create_client_profile_permission(self, user_label: str, should_succeed:
             )
             self.assertEqual(client_count, ClientProfile.objects.count())
 
+    @parametrize(
+        "user_label, should_succeed",
+        [
+            ("org_1_case_manager_1", True),
+            ("org_1_case_manager_2", True),
+            ("org_2_case_manager_1", True),
+            ("client_user_1", False),  # Non CM should not succeed
+            (None, False),  # Anonymous user should not succeed
+        ],
+    )
+    def test_update_client_profile_permission(self, user_label: str, should_succeed: bool) -> None:
+        self._handle_user_login(user_label)
+        original_nickname = ClientProfile.objects.get(id=self.client_profile_1["id"]).nickname
+
+        variables = {"id": self.client_profile_1["id"], "nickname": "updated nick"}
+        response = self._update_client_profile_fixture(variables)
+        updated_nickname = ClientProfile.objects.get(id=self.client_profile_1["id"]).nickname
+
+        if should_succeed:
+            self.assertIsNotNone(response["data"]["updateClientProfile"]["id"])
+            self.assertNotEqual(original_nickname, updated_nickname)
+        else:
+            self.assertEqual(len(response["errors"]), 1)
+            self.assertIn(
+                response["errors"][0]["message"],
+                {
+                    "You do not have permission to modify this client.",
+                    "Cannot filter against a non-conditional expression.",
```

Comment:
this is the `TypeError` raised when running `filter_for_user` on a django guardian anonymoususer